undertow/src/main/java/org/wildfly/extension/undertow/deployment/UndertowDeploymentProcessor.java

```diff
@@ -129,12 +129,43 @@ public void deploy(final DeploymentPhaseContext phaseContext) throws DeploymentU
         if (warMetaData == null) {
             return;
         }
-        String hostName = hostNameOfDeployment(warMetaData, defaultHost);
-        processDeployment(warMetaData, deploymentUnit, phaseContext.getServiceTarget(), hostName);
+        String deploymentName;
+        if (deploymentUnit.getParent() == null) {
+            deploymentName = deploymentUnit.getName();
+        } else {
+            deploymentName = deploymentUnit.getParent().getName() + "." + deploymentUnit.getName();
+        }
+        final UndertowService undertowService = (UndertowService) phaseContext.getServiceRegistry().getService(UndertowService.UNDERTOW).getValue();
+        final Set<Server> servers = undertowService.getServers();
```

Comment:
@dmlloyd @stuartwdouglas can you two guys look at the above code/ conversation and give your input.
I think it **should** be fine and work, but do understand and somewhat share @bstansberry 's concerns.

If we do need to establish hard deps, what would be best way to go? have UndertowService depend on all hosts or even, have UndertowServce depend on all servers.
And each server depend on all child hosts it has defined.

That would for sure enforce that all hosts are up, but it feels like huge overkill.